File: ViewController.m
//
//  ViewController.m
//  ingeSDK
//
//  Created by Gustavo Diaz on 3/17/17.
//  Copyright © 2017 Gustavo Diaz. All rights reserved.
//

#import "ViewController.h"
#import "RootParameters.h"
#import "Methods/CreatePaymentMethodToken.h"
#import "Methods/RetrievePaymentMethodToken.h"
#import "Nps.h"
#import "Nodes/CardDetails.h"
#import "Nodes/Billing.h"
#import "MethodResponses/CreatePaymentMethodTokenResponse.h"
#import "Utils/Environments.h"

@interface ViewController ()

@end

@implementation ViewController

-(IBAction)retrievePMT:(id)sender{
    retrieveLabel.text = @"Requesting...";
    
    Nps *nps = [[Nps alloc]initWithEnvironment:SANDBOX];
    nps.merchantId = @"psp_test";
    nps.pspVersion = @"2.2";
    nps.clientSession = @"zqXc2jw3yXYZztlfLMcqjURaqudKz83kFIOlKkZ3UEcT0fRbqCfPzootb6SsTu3O";
    
    
    NSLog(@"%@", [Nps getDeviceFingerPrint]);
    NSLog(@"%@", @"ese era el iovation");
    
    
    [nps retrievePaymentMethodToken:@"2gOwo2esUsfZwFRy2QxqppRN0H0tHZ9v"
                      methodResponse:^(RetrievePaymentMethodTokenResponse* methodResponse, NSError *error) {
                          NSLog(@"%@", [methodResponse responseExtended]);
                          retrieveLabel.text = [methodResponse.cardDetails maskedNumber];
                      }];
    
}

-(IBAction)recachePMT:(id)sender{
    recacheLabel.text = @"Requesting...";
    
    Nps *nps = [[Nps alloc]initWithEnvironment:SANDBOX];
    nps.merchantId = @"psp_test";
    nps.pspVersion = @"2.2";
    nps.clientSession = @"zqXc2jw3yXYZztlfLMcqjURaqudKz83kFIOlKkZ3UEcT0fRbqCfPzootb6SsTu3O";
    
    
    Billing *billingDetailss = [[Billing alloc]init];
    
    billingDetailss.pspPerson.firstName = @"JOHN DOE";
    
    [nps recachePaymentMethodToken:@"BrMcwXgYtHEfgbHsuYF8WJUceD0s0nMc"
                   cardSecurityCode:@"123"
                     billingDetails:billingDetailss
                     methodResponse:^(RecachePaymentMethodTokenResponse *methodResponse, NSError *error) {
                         NSLog(@"%@", [methodResponse responseExtended]);
                         recacheLabel.text = [methodResponse.cardDetails holderName];
                     }];
    
}

-(IBAction)getProduct:(id)sender{
    getProductLabel.text = @"Requesting ...";
    
    Nps *nps = [[Nps alloc]initWithEnvironment:SANDBOX];
    nps.merchantId = @"psp_test";
    nps.pspVersion = @"2.2";
    nps.clientSession = @"zqXc2jw3yXYZztlfLMcqjURaqudKz83kFIOlKkZ3UEcT0fRbqCfPzootb6SsTu3O";
    
    
    [nps getProduct:@"424242"
        postDateTime:@"2016-12-01 12:00:00"
      methodResponse:^(GetIINDetailsResponse *methodResponse, NSError *error) {
          NSLog(@"%@", [methodResponse responseExtended]);
          getProductLabel.text = [methodResponse product];
      }];
}

-(IBAction)sendButton:(id)sender{
    responseLabel.text = @"Requesting...";
    Nps *nps = [[Nps alloc]initWithEnvironment:SANDBOX];
    nps.merchantId = @"psp_test";
    nps.pspVersion = @"2.2";
    nps.clientSession = @"zqXc2jw3yXYZztlfLMcqjURaqudKz83kFIOlKkZ3UEcT0fRbqCfPzootb6SsTu3O";
    
    CardDetails *card = [[CardDetails alloc]init];
    
    card.number = @"4507990000000010";
    card.holderName = @"JOHN DOE";
    card.securityCode = @"123";
    card.expirationDate = @"1909";
    
    Billing *billingDetails = [[Billing alloc]init];
    
    billingDetails.pspPerson.firstName = @"JOHN DOE";
    
    [nps createPaymentMethodToken:card
                   billingDetails:billingDetails
                   methodResponse:^(CreatePaymentMethodTokenResponse* methodResponse, NSError *error) {
                       if(!error){
                           NSLog(@"%@", [methodResponse paymentMethodToken]);
                           responseLabel.text = [methodResponse paymentMethodToken];
                       }
                   }];
 		
}

-(IBAction)getDeviceFingerPrint:(id)sender{
    NSLog(@"%@", [Nps getDeviceFingerPrint]);
}

- (void)viewDidLoad {
    [super viewDidLoad];
    cardNumberField.text = @"4850110000000000";
    expDateField.text = @"1909";
    cvcField.text = @"123";    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: demo/testPod/ViewController.swift
//
//  ViewController.swift
//  testPod
//
//  Created by Gustavo Diaz on 5/18/17.
//  Copyright © 2017 Gustavo Diaz. All rights reserved.
//

import UIKit
import npsSdk

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        //retrievePaymentMethodToken()
        //createPaymentMethodToken()
        //testiovation()
        //recachePaymentMethodToken()
        //getProductDetail()
        //Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    func testiovation() -> Void{
        print (Nps.getDeviceFingerPrint())
    }
    
    func retrievePaymentMethodToken() -> Void {
        let nps = Nps(environment: SANDBOX)!
        nps.merchantId = "psp_test"
        nps.clientSession = "your_session"
        
        nps.retrievePaymentMethodToken("your_token",
                                        methodResponse: {( methodResponse: RetrievePaymentMethodTokenResponse?, error: Error?) -> Void in
            if error == nil{
                print("\(methodResponse?.responseExtended)")
                print(methodResponse?.dictResponse! as Any)
            }
        })
    }
    
    func recachePaymentMethodToken() -> Void {
        let nps = Nps(environment: SANDBOX)!
        nps.merchantId = "psp_test"
        nps.clientSession = "your_session"
        
        let billing = Billing()
        billing.pspPerson.firstName = "JOHN DOE"
        
        
        nps.recachePaymentMethodToken("your_pmid", cardSecurityCode: "123", billingDetails: billing,
                                      methodResponse: {( methodResponse: RecachePaymentMethodTokenResponse?, error: Error?) -> Void in
                if error == nil{
                    print(methodResponse?.responseCod as Any)
                }
        })
    }
    
    func getProductDetail() -> Void{
        let nps = Nps(environment: SANDBOX)!
        nps.merchantId = "psp_test"
        nps.clientSession = "fuck_session"
        
        nps.getProduct("424242",
                       methodResponse:{( methodResponse: GetIINDetailsResponse?, error: Error?) -> Void in
            if error == nil{
                print(methodResponse?.responseCod as Any)
            }
        })
        
    }
    
    func createPaymentMethodToken() -> Void {
        let nps = Nps(environment: SANDBOX)!
        nps.merchantId = "psp_test"
        nps.clientSession = "your_session"
        
        let card = CardDetails()
        
        card.number = "4507990000000010";
        card.holderName = "JOHN DOE";
        card.securityCode = "123";
        card.expirationDate = "1909";
        
        
        let billing = Billing()
        
        billing.pspPerson.firstName = "JOHN";
        billing.pspPerson.lastName = "Smith";
        billing.pspPerson.dateOfBirth = "1987-01-01";
        billing.pspPerson.gender = "M";
        billing.pspPerson.nationality = "ARG";
        billing.pspPerson.idType = "200";
        billing.pspPerson.idNumber = "95665091";
        billing.pspPerson.phoneNumber1 = "4123-1234";
        billing.pspPerson.phoneNumber2 = "4123-5678";
        
        billing.pspAddress.additionalInfo = "JOHN";
        billing.pspAddress.city = "Buenos Aires";
        billing.pspAddress.stateProvince = "CABA";
        billing.pspAddress.country = "ARG";
        billing.pspAddress.zipCode = "1425";
        billing.pspAddress.street = "suipacha";
        billing.pspAddress.houseNumber = "32123123";
        
        nps.createPaymentMethodToken(card, billingDetails: billing, methodResponse: {( methodResponse: CreatePaymentMethodTokenResponse?, error: Error?) -> Void in
            if error == nil{
                print(methodResponse?.responseExtended as Any)
            }
        })

        
    }
}

